plugins/metrics/src/test/java/org/apache/cloudstack/response/HostMetricsResponseTest.java

```diff
@@ -56,4 +56,15 @@ public void testSetCpuAllocatedWithValidCpu() {
         Assert.assertEquals(expected, hostResponse.getCpuAllocatedGhz());
     }
 
+    @Test
+    public void testSetCpuAllocatedWithNullCpu() {
+        String expected = null;
+        final HostMetricsResponse hostResponse = new HostMetricsResponse();
+        hostResponse.setCpuAllocated(null, 10, 1000L);
+        Assert.assertEquals(expected, hostResponse.getCpuAllocatedGhz());
+        hostResponse.setCpuAllocated(String.format("50%s25%%", decimalSeparator), null, 1000L);
+        Assert.assertEquals(expected, hostResponse.getCpuAllocatedGhz());
+        hostResponse.setCpuAllocated(String.format("50%s25%%", decimalSeparator), 10, null);
+        Assert.assertEquals(expected, hostResponse.getCpuAllocatedGhz());
+    }
```

Comment:
Each unit test should test only one test case. Please break this into different tests.